src/js/msp/MSPHelper.js

```diff
@@ -164,6 +164,16 @@ MspHelper.prototype.process_data = function(dataHandler) {
                     FC.MOTOR_OUTPUT_ORDER[i] = data.readU8();
                 }
                 break;
+            case MSPCodes.MSP2_GET_SMARTAUDIO_DEVICE:
+                FC.SMARTAUDIO_DEVICE = {};
```

Comment:
I'm not too sure if we must initialize here the object. This must be done in the reset function.
When the MSP command grows, and it has conditional parts with the version, this will make the conditional parts empty, and our objective is that they have the "default" value of the reset part.

Reply:
@McGiverGim I added initialization in fc.js `resetState`, it's there. So this line is useless if  `resetState` being called before the MSP command. But isn't it safer to also make sure that the object is created before filling it? For the far future when we can just call MSP commands without pre initialization :)

Reply:
If I'm not wrong, `resetState` is called when disconnecting or connecting, or both. I'm not too sure.

Usually this is not a problem, the values are always returned or not depending on the version, but they are always the same elements (with different values). But your case is a little different, they can come or not depending on the battery powering it or not.

I like your solution, is a better and cleaner code. But if in a future this command grows and some parts are conditional, we will not have the default values in the members of the object. We will have some of them undefined, and can cause problems in the tab.

Maybe the good solution is to call a reset of only the object that is going to be used in each object in the MSP command, but this will imply change the way that it is being done until now.

@mikeller what do you think about this?

Reply:
@McGiverGim This is how i planned, but please correct me if it's wrong and should be done differently:
when VTX has not powered  up or not connected the firmware returns some defaults values:
```
version: 0,
mode: 0,
```
it converts to an unknown SA version in UI

When an old Betaflight connected, MSP command will not get a proper response, and then the default Configurator values will be used. still the same
```
version: 0,
mode: 0,
```
and they will be converted to "Unknown" again.